cart/views.py

```diff
@@ -1,3 +1,92 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.views     import View
+from django.http      import JsonResponse
+
+from .models   import Cart
+
+from utils import login_decorator
+
+class CartListView(View):
+
+    @login_decorator
+    def post(self,request):
+        try:
+            data = json.loads(request.body)
+            user = request.user
+
+            if data['count'] <= 0:
+                return JsonResponse({"message": "LESS_THAN_MINIMUM_QUANTITY"}, status=400) 
+
+            if Cart.objects.filter(user=user, ingredient_id=data['ingredient_id']).exists():
+                cart = Cart.objects.get(user=user, ingredient_id=data['ingredient_id'])
+                cart.count += data['count']
+                cart.save()
+                return JsonResponse({'message':'ADDITIONAL_SUCCESS'}, status=200) 
+                    
+            Cart.objects.create(
+                count         = data['count'],
+                ingredient_id = data['ingredient_id'],
+                user          = user
+                )
+            return JsonResponse({'message':'SUCCESS'}, status=201)
+        
+        except KeyError:
+            return JsonResponse({"message": "KEY_ERROR"}, status=400)
+        
+        except Cart.DoesNotExist:
+            return JsonResponse({'message': 'THIS_ACCOUNT_DOES_NOT_EXIST'}, status=400)
+    
+    @login_decorator
+    def get(self,request):
+        try:
+            users_ingredient_list = Cart.objects.filter(user=request.user)
+            cart_list = []
+
+            for cart in users_ingredient_list:
+                cart_info = {
+                    'id'       : cart.ingredient.id,
+                    'name'     : cart.ingredient.name,
+                    'price'    : cart.ingredient.price,
+                    'image_url': cart.ingredient.image_url,
+                    'count'    : cart.count
+                    }
+                cart_list.append(cart_info)
+            return JsonResponse({'cart_list':cart_list}, status=200)
+
+        except KeyError:
+            return JsonResponse({"message": "KEY_ERROR"}, status=400)
+
+        except Cart.DoesNotExist:
+            return JsonResponse({'message': 'THIS_ACCOUNT_DOES_NOT_EXIST'}, status=400)
+
+    @login_decorator
+    def delete(self,request):
+        try:
+            data = json.loads(request.body)
+            user = request.user
+            
+            [Cart.objects.filter(user_id=user.id, ingredient_id=data_i['ingredient_id']).delete() for data_i in data]
+            return JsonResponse({"message": "SUCCESSFUL_DELETION"}, status=200)
```

Comment:
이경우에 지우고자하는 cart_id나 ingredient_id 리스트만 받아오면 됩니다.
`input : [1, 2, 3]`
 
```suggestion
            user = request.user
            
            carts = Cart.objects.filter(user_id=user.id, ingredient_id__in = data["ingredient_ids"]
            carts.delete()
            
            return JsonResponse({"message": "NO_CONTENT"}, status=204)
```

Reply:
수정완료했습니다.